### PR TITLE
MH-12726 Add description to theme

### PR DESCRIPTION
**Pull request by doofy**
- add theme description to new series wizard
- add theme description to series details
- add necessary endpoints to gather theme descriptions
- a few whitespace fixes

I suggest testing this patch before commit 839b1d0 as creating a series is broken until #95 is merged.

As this is a change to UI here is a screenshot showing before and after the change. The view in series details has the same change. 

[create series before](https://i.imgur.com/Ue8JBBK.png)
[create series after](https://i.imgur.com/19BcKit.png)

This work is sponsored by SWITCH.

### CODE REVIEW

**Comment by JulianKniephoff** [comment]
At the end of the new series wizard, a raw JSON is visible in the UI:

image

**Comment by JulianKniephoff** [comment]
When selecting the theme in the new series wizard I see the description as you showed iin your screenshots, but in the series details it only shows up after selecting a different theme, when the series was created with a theme to begin with.

Proof: 
image

File: modules/admin-ui/src/main/resources/public/org/opencastproject/adminui/languages/lang-en_US.json
Comment (JulianKniephoff) [inline modules/admin-ui/src/main/resources/public/org/opencastproject/adminui/languages/lang-en_US.json]: Why not … (i.e. U+2026)?

Also many style guides suggest separating ellipses from the surrounding text with a space.

File: modules/admin-ui/src/main/java/org/opencastproject/adminui/endpoint/SeriesEndpoint.java
Comment (JulianKniephoff) [inline modules/admin-ui/src/main/java/org/opencastproject/adminui/endpoint/SeriesEndpoint.java]: `list` is not the most descriptive variable name, especially since this is an object, not a list. How about `theme` (or `themeJson` for consistency …)

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/series-details.html
Comment (JulianKniephoff) [inline modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/series-details.html]: ~~availabel~~available

Also this comment is not in title case like the others. :wink:

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-series.html
Comment (JulianKniephoff) [inline modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-series.html]: See above

File: modules/admin-ui/src/main/webapp/scripts/modules/events/controllers/serieController.js
Comment (JulianKniephoff) [inline modules/admin-ui/src/main/webapp/scripts/modules/events/controllers/serieController.js]: Can we call this `selectedTheme` (camel case) for better readability?

File: modules/admin-ui/src/main/webapp/scripts/modules/events/controllers/serieController.js
Comment (JulianKniephoff) [inline modules/admin-ui/src/main/webapp/scripts/modules/events/controllers/serieController.js]: Same here: `themeDescriptions` would be preferable, imo.

File: modules/admin-ui/src/main/webapp/scripts/modules/events/controllers/serieController.js
Comment (JulianKniephoff) [inline modules/admin-ui/src/main/webapp/scripts/modules/events/controllers/serieController.js]: `selectedThemeId`

File: modules/admin-ui/src/main/webapp/scripts/modules/events/controllers/serieController.js
Comment (JulianKniephoff) [inline modules/admin-ui/src/main/webapp/scripts/modules/events/controllers/serieController.js]: The JS-nerd in me wants to replace this with a simple `== null`, but I see why one would want to avoid explicit use of `==`. :disappointed:

File: modules/admin-ui/src/main/webapp/scripts/modules/events/controllers/serieController.js
Comment (JulianKniephoff) [inline modules/admin-ui/src/main/webapp/scripts/modules/events/controllers/serieController.js]: If you are touching these lines anyway, couldn't we add spaces inside the `{}` for consistency? Or is that inconsistent anyway? I saw it with spaces somewhere in this PR.

File: modules/admin-ui/src/main/webapp/scripts/modules/events/controllers/serieController.js
Comment (JulianKniephoff) [inline modules/admin-ui/src/main/webapp/scripts/modules/events/controllers/serieController.js]: See above.

File: modules/admin-ui/src/main/webapp/scripts/shared/services/wizards/new-series/theme.js
Comment (JulianKniephoff) [inline modules/admin-ui/src/main/webapp/scripts/shared/services/wizards/new-series/theme.js]: `themeDescription`

File: modules/admin-ui/src/test/java/org/opencastproject/adminui/endpoint/SeriesEndpointTest.java
Comment (JulianKniephoff) [inline modules/admin-ui/src/test/java/org/opencastproject/adminui/endpoint/SeriesEndpointTest.java:292]: ~~There is a space in front of a comma missing here.~~

I see now. Changing this would require changing the actual response as well, which might not be worth it (or undesirable, even).

File: modules/admin-ui/src/test/resources/test/unit/modules/events/controllers/serieControllerSpec.js
Comment (JulianKniephoff) [inline modules/admin-ui/src/test/resources/test/unit/modules/events/controllers/serieControllerSpec.js]: Should there be spacing in the object literal as in the line above for consistency?

**Review by JulianKniephoff — CHANGES_REQUESTED** [review]

File: modules/admin-ui/src/main/resources/public/org/opencastproject/adminui/languages/lang-en_US.json
Comment (lkiesow) [inline modules/admin-ui/src/main/resources/public/org/opencastproject/adminui/languages/lang-en_US.json]: This text is not correct. The themes can be applied at any time during a workflow execution or not at all. It is a configurable operation. It certainly may not only happen when publishing material.

File: modules/admin-ui/src/main/resources/public/org/opencastproject/adminui/languages/lang-en_US.json
Comment (lkiesow) [reply modules/admin-ui/src/main/resources/public/org/opencastproject/adminui/languages/lang-en_US.json]: +1 for using `…`. We already use an ellipsis in most strings. There was a time when it was used for all strings :-(

> Also many style guides suggest separating ellipses from the surrounding text with a space.

And others don't. Until now, a space character is used nowhere in Opencast and would rather break consistency.

Comment (lkiesow) [reply modules/admin-ui/src/main/resources/public/org/opencastproject/adminui/languages/lang-en_US.json]: Looking at this and other, similar strings, I would probably omit the ellipsis completely and just say `Select theme` or `Select a theme`

**Review by lkiesow — CHANGES_REQUESTED** [review]

File: modules/admin-ui/src/main/resources/public/org/opencastproject/adminui/languages/lang-en_US.json
Comment (JulianKniephoff) [reply modules/admin-ui/src/main/resources/public/org/opencastproject/adminui/languages/lang-en_US.json]: @lkiesow the other style guides are obviously wrong! :stuck_out_tongue_closed_eyes: 

Seriously though, consistency > style (within reason) and I would actually prefer the version without ellipsis that you braught up, too. :thinking:

**Comment by doofy** [comment]
@JulianKniephoff thanks for the great review so far. I pushed some fixes. Please continue your review.

File: modules/admin-ui/src/main/webapp/scripts/modules/events/controllers/serieController.js
Comment (JulianKniephoff) [inline modules/admin-ui/src/main/webapp/scripts/modules/events/controllers/serieController.js:213]: Not sure about the spacing here. I definitely have seen both, `if(` and `if (`, but the latter seems prevalent.

File: modules/admin-ui/src/main/webapp/scripts/modules/events/controllers/serieController.js
Comment (JulianKniephoff) [inline modules/admin-ui/src/main/webapp/scripts/modules/events/controllers/serieController.js:212]: I am no Angular expert, but I see this function called only within this very file. Is there a reason why this lives in the `$scope`? I personally am a friend of free functions for cases like this one, but there might be some Angular best practices/idiosyncrasies I don't know about. Or maybe it is just consistency, again ...? :thinking:

**Review by JulianKniephoff — APPROVED** [review]

**Comment by JulianKniephoff** [comment]
It works now and I also understand the code better now, so that I can say that its totally reasonable.

I added two further nitpicks, but they are very minor and I am not even that sure about them myself. I'd leave the decision to change these to @doofy or whoever just wants to merge this, hence the seemingly contradictory review with nagging comments, but with an overall approval. :wink:

**Comment by doofy** [comment]
@JulianKniephoff thanks for you final review. I say this request is ready for merging.

**Comment by staubesv** [comment]
Hi @JulianKniephoff, please either merge or ask for changes ;-)

**Comment by lkiesow** [comment]
@staubesv, Julian is no committer and hence cannot merge this pull request.
He asked me to take a look at this. But if you want to, please go ahead and merge this.

**Comment by staubesv** [comment]
@lkiesow Did some tests and seems to work -> I will merge this.

**Pull request merged**
